customerrors/error.go

```diff
@@ -18,33 +18,20 @@ package customerrors
 
 import (
 	"fmt"
-	"os"
-
-	"github.com/sonatype-nexus-community/nancy/buildversion"
-	. "github.com/sonatype-nexus-community/nancy/logger"
 )
 
-type SwError struct {
-	Message string
-	Err     error
-}
-
-func (sw SwError) Error() string {
-	return fmt.Sprintf("%s - error: %s", sw.Message, sw.Err.Error())
-}
-
-func (sw SwError) Exit() {
-	os.Exit(3)
+type ErrorExit struct {
+	Message  string
+	Err      error
+	ExitCode int
 }
 
-func Check(err error, message string) {
-	if err != nil {
-		location, _ := LogFileLocation()
-		myErr := SwError{Message: message, Err: err}
-		LogLady.WithField("error", err).Error(message)
-		fmt.Println(myErr.Error())
-		fmt.Printf("For more information, check the log file at %s\n", location)
-		fmt.Println("nancy version:", buildversion.BuildVersion)
-		myErr.Exit()
+func (ee ErrorExit) Error() string {
+	var errString string
+	if ee.Err != nil {
+		errString = ee.Err.Error()
+	} else {
+		errString = "nil"
```

Comment:
Why `"nil"`? There is no such thing as a nil string, why not `""`?

Reply:
Good point. Changed to "".  ccc94e1